### PR TITLE
[sdl2] Update to 2.30.6

### LINKED ISSUE #40338: [SDL2] update to 2.30.6
### Library name

SDL2

### New version number

2.30.6

### Other information that may be useful (release notes, etc...)

This is a stable bugfix release, with the following changes:
- Improved detection of Nintendo Switch Pro controller report mode
- Fixed a rare crash when a controller is disconnected
- Fixed creating a framebuffer with KMSDRM on some systems

### PR DESCRIPTION
**Pull request by AregevDev**
~~Fixes https://github.com/microsoft/vcpkg/issues/40338
Added a patch to fix the Android ARM64 build temporarily. The fix is already merged into SDL2 master, it will be removed in the next port update.~~
Waiting for the next SDL release.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by bwrsandman** [comment]
You might want to include this patch https://github.com/libsdl-org/SDL/commit/558630d59c2db73ac7c0c0cae000e4bee6ef42cf.patch in order to avoid #40268

**Comment by AregevDev** [comment]
Is this fix still on master? Can't it wait for SDL's next release?

**Comment by bwrsandman** [comment]
It's merged into master of sdl2 but it's a compilation error for that release so in any case it will need a patch.

**Comment by AregevDev** [comment]
I see

**Comment by AregevDev** [comment]
I applied the patch and SDL2 builds properly on arm64... I don't have a way to test it in an application though. I'll need to remove this patch in the next SDL2 release, right?

**Comment by bwrsandman** [comment]
> I'll need to remove this patch in the next SDL2 release, right?
Yes most likely

I tested that it builds with nkd r27 and it does with the patch.

File: versions/s-/sdl2.json [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "056c66809f624a18cbb4870ce3d01f47757014d8",
+      "version": "2.30.6",
+      "port-version": 0
+    },
+    {
+      "git-tree": "9aa8e3cb992ced7629d28bec2d7fe8ab28204d42",
+      "version": "2.30.6",
+      "port-version": 1
+    },
```
Comment (jimwang118) [inline versions/s-/sdl2.json:12]:
```suggestion
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

**Review by jimwang118 — DISMISSED** [review]

**Comment by jimwang118** [comment]
All features passed with following triplets:

```
x64-windows
x64-windows-static
```

**Comment by data-queue** [comment]
I'm not sure about the patch: the upstream patch doesn't apply cleanly since it seems the code around it has refactored significantly. I think it's better to either wait for a new `sdl2` release or just not fix the android NDK issue in this PR.

**Comment by bwrsandman** [comment]
I can make a separate PR for the patch

**Comment by AregevDev** [comment]
Yeah, sounds good. I'll revert the changes when I can.

**Comment by AregevDev** [comment]
Do I need to add a `port-version`?

**Comment by erikyuzwa** [comment]
@AregevDev - here's the PR for `2.30.5` - https://github.com/microsoft/vcpkg/pull/39710 - maybe that helps?

**Comment by dg0yt** [comment]
> Do I need to add a `port-version`?

No (for `ports/sdl2/vcpkg.json`). Revision Zero of a new version doesn't need it.

**Comment by AregevDev** [comment]
Seems ok now...

**Review by data-queue — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**